source/common/router/router.h

```diff
@@ -138,6 +145,32 @@ class Filter : Logger::Loggable<Logger::Id::router>,
     return callbacks_->connection();
   }
 
+  /**
+   * Set a pseudo-random cookie with the name @param key and TTL @param max_age
+   * to be sent to the downstream host.
+   * @return std::string the value of the new cookie
+   *
+   * marked const so it can be called from hashKey(), and because all mutated
```

Comment:
I'm not seeing much use of mutable in the envoy code base, and generally for things like stats.
I'm having trouble sorting this out without patching and compiling but would it be possible to instead pass in a list downstream_set_cookies in as an argument rather than accessing it as a member variable, so the member doesn't have to be mutable?

Reply:
After talking in-person, I realized that there's no way to avoid having `mutable` somewhere. The call graph that hits `addDownstreamSetCookie()` always goes through `Filter::hashKey()`, which is itself const. Passing a mutable list of cookies in would also work but `hashKey()` would then be the mutating function masquerading as const.

Reply:
Maybe we should change `hashKey()` to be non-const? While conceptually it starts out as a simple computational hash, if we're going to start modifying the actual request, let's just be upfront about it I reckon.